templates/powercli_esx_ntp_onlyif.ps1.erb

```diff
@@ -17,8 +17,15 @@ foreach($s in $NTPServersArray){
     # Trim the whitespace from the current $s.
     # This is to remove the space that starts ' 192.168.1.11'
     $s = $s.Trim(" ")
-    
-    # If the NTP server we wanted ISN'T in the $NTPServersCheck array, queue puppet to set NTP servers
+
+    # If the host had no NTP servers set, we cannot check if it 'contains' the NTP servers
+    # we want. This loop causes puppet to run if the host had NO NTP servers set.
+    if(!$NTPServersCheck){
```

Comment:
you can just include this in the if below:
```
if (!$NTPServersCheck -or !$NTPServersCheck.Contains($s)) {
```
or something like that, https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_logical_operators?view=powershell-6